website/src/useHashRouter.ts

```diff
@@ -3,6 +3,27 @@ import { actions, RootState } from './state';
 import { useDispatch, useSelector } from 'react-redux';
 import { webhookImplementation } from './webhook.impl';
 
+async function encodeState(state: any): Promise<string> {
+    const stream = new Blob([JSON.stringify(state)], {
+        type: 'application/json',
+    }).stream();
+    const compressedReadableStream = stream.pipeThrough(new CompressionStream('gzip'));
+
+    const buffer = await new Response(compressedReadableStream).arrayBuffer();
```

Comment:
This method may result in a runtime error in some browsers when the ArrayBuffer is large.

Maybe we should do a loop as pointed out in https://stackoverflow.com/questions/9267899/how-can-i-convert-an-arraybuffer-to-a-base64-encoded-string#comment16530117_11562550

Reply:
[`8df3ebb` (#60)](https://github.com/StartITBot/discord.builders/pull/60/commits/8df3ebbbbcdeda8a9927d8245e9ab3759f4695ed)

Done!